runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -139,28 +139,36 @@ private Layout getLayoutUninterruptibly() {
                     }
                     return rLayout;
                 }));
+            } catch (ServerNotReadyException se) {
+                log.warn("Server still not ready. Waiting for server to start "
+                        + "accepting requests.");
+                if (rethrowAllExceptions) {
+                    throw se;
+                }
+            } catch (WrongEpochException we) {
+                log.warn("Got a wrong epoch exception, updating epoch to {} and "
+                        + "invalidate view", we.getCorrectEpoch());
+                if (rethrowAllExceptions) {
+                    throw we;
+                }
+            } catch (NetworkException ne) {
+                log.warn("layoutHelper: System seems unavailable", ne);
+                if (rethrowAllExceptions) {
+                    throw ne;
+                }
+            } catch (WrongClusterException wce) {
+                log.warn("layoutHelper: Cluster reconfiguration or incorrect cluster", wce);
+                log.info("layoutHelper: Invoking the systemDownHandler.");
+                runtime.getParameters().getSystemDownHandler().run();
+                throw wce;
             } catch (RuntimeException re) {
                 if (re.getCause() instanceof TimeoutException) {
                     log.warn("Timeout executing remote call, invalidating view and retrying "
                             + "in {}s", retryRate);
```

Comment:
Can this TimeoutException also be caught outside similar to others?
Then, in the remaining space of this catch clause, maybe we can leave a warning message saying "`layoutHelper: An unknown RuntimeException occurred.`" and throw it back to the caller.

Reply:
The TimeoutException seems to be wrapped within the RuntimeException. So it's not possible to catch it directly.
Although can add the unknown exception log in the else block.